components/director/internal/domain/runtime/service.go

```diff
@@ -337,7 +360,39 @@ func (s *service) ListLabels(ctx context.Context, runtimeID string) (map[string]
 		return nil, errors.Wrap(err, "while getting label for Runtime")
 	}
 
-	return labels, nil
+	return s.extractNotProtectedLabels(labels, s.protectedLabelPattern)
+}
+
+func (s *service) extractNotProtectedLabels(labels map[string]*model.Label, protectedLabelsKeyPattern string) (map[string]*model.Label, error) {
+	result := make(map[string]*model.Label)
+	for labelKey, label := range labels {
+		if protected, err := s.isProtected(labelKey, protectedLabelsKeyPattern); err != nil {
```

Comment:
I got a bit confused by this inline variable initialization and the error check at the same time because the "if-else if" construction here is somewhat logically strange.

I can suggest a cleaner approach but this is just a personal preference of mine.

```suggestion
protected, err := s.isProtected(labelKey, protectedLabelsKeyPattern)
if err != nil {
    return nil, err
} 

if !protected {
    result[labelKey] = label
}
```

Note: this is also valid for `func (s *service) unsafeExtractNotProtectedLabels`